kufar/books/main.go

```diff
@@ -334,31 +340,45 @@ func ReadBooks(d Database) func(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-//Get gets book by id
-func (c *cache) Get(id int) (Book, bool) {
-	c.mu.RLock()
-	defer c.mu.RUnlock()
+func (*rediscache) Get(id string) (Book, error) {
+	c := newClient()
```

Comment:
You don't need to create a new client here. Just use an existing object here. Same for `Set` & `Remove`